### ISSUE TITLE
Expose `gilrs` directly

### ISSUE THREAD

**Issue by aleokdev** [ggez/ggez#1026]
I ran into this issue while trying to implement `Deserialize` for a type that contained a `GamepadId`.
The current `gilrs` wrapper (src/input/gamepad.rs) does not include more functionality nor does it provide any meaningful change to its interface. I suggest removing it entirely and exposing `gilrs` as-is, since the wrappers hurt the featureset of the crate by artificially limiting what is possible to do with it.

An alternative would be to just make all of their fields `pub`. I don't see any reason why they should be crate-protected.

**Comment by PSteinhaus** [comment]
I agree that its fields not being `pub` artificially limits and hurts the featureset of `gilrs`. I wonder why we didn't change this sooner. After all, exposing internals to allow users to leverage their full functionality, if necessary, is a bit of a theme in ggez, with exposing the graphics context, showing how to build your own event loop and such...

Looking at the `gamepad` module I think removing it might really be a good idea. There might be _some_ merit in it, if we removed the unnecessary wrappers and instead added functions like `is_pressed(&ctx.gilrs, id, button)` and `just_pressed(&ctx.gilrs, id, button)` to easily check whether a button is currently pressed down and whether it has just been pressed, making both a bit easier [than without](https://docs.rs/gilrs/latest/gilrs/struct.Gilrs.html#counter) (though I admit that in the first case it's just `gilrs.gamepad(id).is_pressed(button)` vs. `is_pressed(&ctx.gilrs, id, button)`, which isn't necessarily an improvement, but if we provided `just_pressed`, which I think is useful, then we should also provide `is_pressed` for consistency).

Then again, that would be a module level function, instead of a context method. But if we wanted that, then we'd have to live with the wrapper, changed to publicly expose the `gilrs` context it holds.
